src/app/Components/recipe-detail/recipe-detail.component.ts

```diff
@@ -0,0 +1,39 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { RecipeService } from 'src/app/Services/recipe.service';
+
+@Component({
+  selector: 'app-recipe-detail',
+  templateUrl: './recipe-detail.component.html',
+  styleUrls: ['./recipe-detail.component.scss'],
+})
+export class RecipeDetailComponent implements OnInit {
+  recipeDetail: any;
+  toggle = false;
+
+  constructor(
+    private dataRecipe: RecipeService,
+    private route: ActivatedRoute,
+    private router: Router
+  ) {}
+
+  ngOnInit(): void {
+    this.route.params.subscribe((param) => {
+      const id = Number(param['id']);
+      this.dataRecipe.recipes.subscribe((res) => {
```

Comment:
don't implement nested subscribe. use a combination operator of RxJs to do only 1 subscription.